### PR TITLE
CI pipeline for checks on PR creation and branch pushes

### PR DESCRIPTION
**Pull request by mmerklinger**
This PR adds a CI pipeline to run checks on pushes to non-master branches and pull request.

## Changes

- Adds a Github workflow file.
- Restructures check and fix targets in the Makefile.
- Moves check configuration to `.flake8` or `pyproject.toml` configuration files.
- Changes pull request template to reflect the change in the process.

## Checklist

- [ ] tested with Python3.9
- [ ] run `make check` or `make fix` for the formatting check
- [x] signed commits
- [ ] updated documentation (e.g. parameter description, inline doc, docs.nitrokey)
- [ ] added labels

## Test Environment and Execution

- OS:
- device's model:
- device's firmware version:

### Relevant Output Example

Fixes #

### CODE REVIEW

**Comment by daringer** [comment]
this is the path towards github built `.msi` and `.exe` releases

**Comment by mmerklinger** [comment]
> this is the path towards github built `.msi` and `.exe` releases

@daringer This PR will only provide the CI pipeline. The CD pipeline will follow in another PR, after a bit more testing. Hence, this PR is complete and ready for review.

**Comment by daringer** [comment]
please have a short look here @szszszsz & @robin-nitrokey ... `0.4.35` is now released, I would suggest we target `0.5` very soon(tm) including this github CI pipeline and the matching release pipeline #361

**Review by robin-nitrokey — COMMENTED** [review]
Looks good to me.  Why are the checks split into separate jobs though?

File: Makefile [hunk 1]
```diff
@@ -33,27 +29,30 @@ rune:
 builde:
 	earthly +build
 
-# ensure this passes before committing
-check: lint
+# code checks
+check-format:
+	$(PYTHON3) -m black --check $(PACKAGE_NAME)/
```
[the remark this answers is not in the document]
Comment (mmerklinger) [reply Makefile:34]: I didn't want to have hard coded paths in the targets. Hence, I changed it to the existing `$(PYTHON3)` variable. If we want to use the path, we should change it there I think.

**Comment by mmerklinger** [comment]
> Looks good to me. Why are the checks split into separate jobs though?

They are separate for three reasons:
1. With the current implementation you know the name of the check from the job name if one fails.
2. The error log can be long and hard to read when you have multiple failing checks in one job.
3. Separate jobs can run in parallel, which makes the feedback faster.

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (robin-nitrokey) [reply Makefile:34]: Hmm, I think `$PYTHON3`is used in some targets to refer to the system binary.  What do you think about adding a `PYTHON3_VENV` variable for the check targets?

**Comment by robin-nitrokey** [comment]
As far as I see, the setup of the environment is the same for all tasks and takes much longer than actually running the tests, so I would default to keeping them in the same job.  But I don’t have a strong opinion on it.

**Review by robin-nitrokey — APPROVED** [review]

File: Makefile (hunk 1 above)
[the remark this answers is not in the document]
Comment (mmerklinger) [reply Makefile:34]: Sounds like a good solution! I will implement this...

**Comment by mmerklinger** [comment]
> As far as I see, the setup of the environment is the same for all tasks and takes much longer than actually running the tests, so I would default to keeping them in the same job. But I don’t have a strong opinion on it.

I would vote for keeping separate jobs mostly because of the number 1 and 2 of my reasoning mentioned above. Your argument about the time is valid. I was planning to look into using a cache to speed this up. I wanted this to test before though.

**Comment by daringer** [comment]
> I was planning to look into using a cache to speed this up. I wanted this to test before though.

This should be the way to go to fulfill all mentioned requirements. 

Let's merge this and #361 so we could also allow our customers to use some of the not officially released builds to update all Nitrokey 3 variants to v1.3.

**Pull request merged**